rs/types/types/src/consensus/idkg.rs

```diff
@@ -65,6 +65,84 @@ pub enum CompletedSignature {
     Unreported(crate::batch::ConsensusResponse),
 }
 
+/// A [`MasterPublicKeyId`], that contains a variant that is compatible with the IDKG protocol.
+///
+/// The [`MasterPublicKeyId`] can hold a number of different key types.
+/// Some of them can be used with the IDKG protocol, while others can not.
+/// The [`IDkgMasterPublicKeyId`] type indicates, that this key id can be used with a IDKG protocol.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+#[cfg_attr(test, derive(ExhaustiveSet))]
+pub struct IDkgMasterPublicKeyId(MasterPublicKeyId);
+
+impl TryFrom<MasterPublicKeyId> for IDkgMasterPublicKeyId {
+    type Error = String;
+
+    fn try_from(val: MasterPublicKeyId) -> Result<Self, Self::Error> {
+        if !val.is_idkg_key() {
+            Err("This key is not an idkg key".to_string())
+        } else {
+            Ok(Self(val))
+        }
+    }
+}
```

Comment:
I guess an alternative would be to introduce a new enum that only has the ECDSA and Schnorr variants, and then do the conversion based on that. It would maybe help us to avoid some of these situations:
```rust
match key_id.inner() {
    MasterPublicKeyId::Ecdsa(_) => [...],
    MasterPublicKeyId::Schnorr(_) => [...],
    MasterPublicKeyId::VetKd(_) => panic!("not applicable to vetKD"),
}
```
I'm not sure if it makes a huge difference otherwise

Reply:
Yes, this was an alternative that I considered. 

The upside is that we don't have these weird leftover matches. The downside is that we cannot implement `Deref<Target=MasterPublicKeyId>`, which is useful because you can pass a `&IdkgMasterPublicKeyId` as a `&MasterPublicKeyId`.

I decided to go this way, we can actually change the type later without touching non-consensus code, if we change our mind.

Reply:
I see, that make sense

Reply:
One could combine the approaches, to get the best of both worlds:
```
struct IDkgMasterPublicKeyId {
    inner: MasterPublicKeyId,
    kind: IDkgMasterPublicKeyIdType,
}

enum IDkgMasterPublicKeyIdType {
    Ecdsa(EcdsaKeyId),
    Schnorr(SchnorrKeyId),
}

impl IDkgMasterPublicKeyId {
    pub fn inner(&self) -> &MasterPublicKeyId {
        &self.inner
    }

    fn kind(&self) -> &IDkgMasterPublicKeyIdEnum {
        &self.kind
    }
}
```
and then do
```
match key_id.kind() {
    MasterPublicKeyId::Ecdsa(_) => [...],
    MasterPublicKeyId::Schnorr(_) => [...],
}
```

Reply:
Hmm, but then we have two clones of the inner key type (if I understand the proposal correctly), which to me feels a bit weird too.

I propose, we leave as is. Once we have implemented VetKeys and we go to the "cleanup" part, we see how important that `Deref` implementation really is. If we barely ever use it, we could instead replace this implementation with the enum variant. What do you think?

Reply:
That is indeed a downside of the combined approach. If Rust supported self-referential structs, then we could use a reference to the key ID in `IDkgMasterPublicKeyIdType`, but it doesn't. One thing that would somewhat improve the situation would be if `EcdsaKeyId` and `SchnorrKeyId` were `Copy` (which we actually might do at some point to revert https://github.com/dfinity/ic/pull/2347).

Anyway, I'm OK if you prefer to leave it as is.

Reply:
Well, we could also make the IDkgMasterPublicKeyIdType have borrowed inner values, since we are only using them when matching.
This is something that I'm going to try, but not as part of this PR, since it is already fairly large.